e-commerce/src/App.js

```diff
@@ -28,157 +29,138 @@ export default function App() {
+        }}
+      );
+      setCart(response.data.cart)
+    } catch (error) {
+      console.log(error);
+    }
   };
 
-  const addToWishList = (product) => {
-    // Call post api
-    fetch("api/user/wishlist", {
-      method: "POST",
-      headers: {
-        authorization: encodedToken,
-      },
-      // Stringify data and send it
-      body: JSON.stringify({ product }),
-    })
-      .then((data) => data.json())
-      .then((data) => setWishList(data.wishList))
-      .catch((err) => {
-        console.log("Error", err);
-      });
+  const addToWishList = async (product) => {
+    try {
+      const response = await axios.post(
+        `/api/user/wishlist`, { product }, {headers: {
+          authorization: encodedToken,
+        }}
+      );
+      setWishList(response.data.wishList)
+    } catch (error) {
+      console.log(error);
+    }
   };
 
-  const removeFromCart = (product) => {
-    fetch("api/user/cart", {
-      method: "DELETE",
-      headers: {
-        authorization: encodedToken,
-      },
-      // Stringify data and send it
-      body: JSON.stringify({ product }),
-    })
-      .then((data) => data.json())
-      .then((data) => setCart(data.cart))
-      .catch((err) => {
-        console.log("Error", err);
-      });
+  const removeFromCart = async (product) => {
+    try {
+      const response = await axios.delete(
+        `/api/user/cart/${product._id}`,  {headers: {
+          authorization: encodedToken,
+        }}
+      );
+      setCart(response.data.cart)
+    } catch (error) {
+      console.log(error);
+    }
   };
-  const removeFromWishList = (product) => {
-    fetch("api/user/wishlist", {
-      method: "DELETE",
-      headers: {
-        authorization: encodedToken,
-      },
-      // Stringify data and send it
-      body: JSON.stringify({ product }),
-    })
-      .then((data) => data.json())
-      .then((data) => setWishList(data.wishList))
-      .catch((err) => {
-        console.log("Error", err);
-      });
+  const removeFromWishList = async(product) => {
+    try {
+      const response = await axios.delete(
+        `/api/user/wishlist/${product._id}`,  {headers: {
+          authorization: encodedToken,
+        }}
+      );
+      setWishList(response.data.wishList)
+    } catch (error) {
+      console.log(error);
+    }
   };
 
-  const incrementQuantityHandler = (product) => {
-    fetch(`api/user/cart/${product._id}`, {
-      method: "POST",
-      headers: {
-        authorization: encodedToken,
-      },
-      // Stringify data and send it
-      body: JSON.stringify({ action:{type: "increment"} }),
-    })
-      .then((data) => data.json())
-      .then((data) => setCart(data.cart))
-      .catch((err) => {
-        console.log("Error", err);
-      });
+  const incrementQuantityHandler = async (product) => {
+    try {
+      const response = await axios.post(
+        `api/user/cart/${product._id}`, { action:{type: "increment"} }, {headers: {
```

Comment:
Suggestion: I think we can send the updated product instead of sending the action as the product quantity is already in the product. 
1. Currently in the backend we are hardcoding that the product quantity will increment or decrement with 1 itself but user can inc or dec by any number (depends on frontend). 
2. You won't need to hardcode the if else which will also decrease the unneccessary code.

Reply:
This is great idea. Would love if @tanaypratap can give insights about the best practice around this. (Can push this in a separate PR and creating a new issue for this).